src/libs/pluginprocess/pluginprocess.c

```diff
@@ -71,18 +106,26 @@ void elektraPluginProcessStart (Plugin * handle, ElektraPluginProcess * pp)
 {
 	Key * commandPipeKey = keyNew ("/pluginprocess/pipe/command", KEY_VALUE, pp->commandPipe, KEY_END);
 	Key * resultPipeKey = keyNew ("/pluginprocess/pipe/result", KEY_VALUE, pp->resultPipe, KEY_END);
-	KeySet * keySet = ksNew (0, KS_END);
 	int counter = 0;
 
 	do
 	{
-		ELEKTRA_LOG_DEBUG ("Child: Wait for a KeySet");
-		elektraInvoke2Args (pp->dump, "get", keySet, commandPipeKey);
-		ELEKTRA_LOG_DEBUG ("Child: We received a KeySet with %zd keys in it", ksGetSize (keySet));
+		KeySet * commandKeySet = ksNew (6, KS_END);
```

Comment:
Why 6? What about sending the size instead of "exists"?

Reply:
commandKeySet WILL contain 6 keys in the end. But i agree that i could send over the payload's size for an optimization.

Reply:
Ahh, my fault. Then 6 is perfect.